pkg/olog/default_handler.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 Outreach Corporation. All Rights Reserved.
+
+// Description: Contains logic for determining which handler should be
+// used by default.
+
+package olog
+
+import (
+	"io"
+	"log/slog"
+	"os"
+	"sync/atomic"
+
+	"golang.org/x/term"
+)
+
+var (
+	// defaultHandler is the global handler used by all loggers returned
+	// by this package. For details on how this is set, see
+	// `determineDefaultHandler`.
+	defaultHandler atomic.Int32
+
+	// defaultOut is the default output for the default handler. This is
+	// set to `os.Stderr` by default.
+	defaultOut = os.Stderr
+)
+
+// DefaultHandlerType denotes which handler should be used by default.
+// This is calculated via the `setDefaultHandler` function on package
+// init.
+type DefaultHandlerType int
+
+const (
+	JSONHandler DefaultHandlerType = iota
+	TextHandler
+)
+
+// determineDefaultHandler sets the default handler based on the current
+// environment. If the `os.Stderr` is a TTY, then the default handler
+// is a `slog.TextHandler`. Otherwise, the default handler is the
+// `slog.JSONHandler`.
+func determineDefaultHandler() {
```

Comment:
This will allow us to automatically pretty-print logs when running in development.